src/Html.php

```diff
@@ -1608,6 +1608,7 @@ public static function address(string|Stringable $content = '', array $attribute
      */
     public static function renderTagAttributes(array $attributes): string
     {
+        $attributes = self::filterNullAttributes($attributes);
```

Comment:
I think a separate loop doesn't work well from performance standpoint. Since we have a loop here already, isn't it better to do checks in it?

Reply:
There aren't many attributes in a tag but there could be many tags per page.

Reply:
Yep, I painfully overcomplicated that.